src/app/ReadClient.cpp

```diff
@@ -693,7 +693,7 @@ CHIP_ERROR ReadClient::ProcessAttributeReportIBs(TLV::TLVReader & aAttributeRepo
             ReturnErrorOnFailure(data.GetDataVersion(&version));
             attributePath.mDataVersion.SetValue(version);
 
-            if (mReadPrepareParams.mpDataVersionFilterList != nullptr)
+            if (mReadPrepareParams.mpDataVersionFilterList != nullptr && mReadPrepareParams.mDataVersionFilterListSize != 0)
```

Comment:
@yunhanw-google Why was this change needed?

Reply:
thanks, it is not needed, but i think it is better to have?

Reply:
Why?